src/Response.php

```diff
@@ -648,7 +648,7 @@ protected function decodeDeflate($body)
          */
         $zlibHeader = unpack('n', substr($body, 0, 2));
 
-        if ($zlibHeader[1] % 31 === 0) {
+        if ($zlibHeader[1] % 31 === 0 && ord($body[0]) === 0x78 && in_array(ord($body[1]), [0x01, 0x5e, 0x9c, 0xda])) {
```

Comment:
Needs tests, IMO.

Possible to hit this through `fromString()`? 

https://github.com/laminas/laminas-http/blob/40218e9786a471832b4e52a9fe900e0b130aa641/src/Response.php#L203